bn-apps/cordapp-updates-distribution/design/design.md

```diff
@@ -0,0 +1,135 @@
+![Corda](https://www.corda.net/wp-content/uploads/2016/11/fg005_corda_b.png)
+
+# CordApp distribution service (CDS)
+
+DOCUMENT MANAGEMENT
+---
+
+## Document Control
+
+| Title                | CordApp distribution service                             |
+| -------------------- | ------------------------------------------------------------ |
+| Date                 | 22 Aug 2018                                                |
+| Author               | Ivan Schasny, Mike Hearn |
+| Distribution         | Design Review Board, Product Management, Solutions Engineering, Platform Delivery |
+| Corda target version | OS                                                   |
+
+## HIGH LEVEL DESIGN
+
+### Overview
+
+This proposal describes the architecture of a reference implementation for a CorDapp Distribution Service (CDS), that would allow BNOs to distribute CorDapps updates to Business Network participants. This design targets scenarios when Business Network members maintain their nodes by themselves and can decide on what version of a CorDapp they would like to upgrade to. This design does *not* target scenarios, when upgrade function is centralised and the whole network can be shut down to perform stop-upgrade-restart.
+
+### Requirements
+
+The following requirements have been gathered from various internal discussions within R3 and from [groups.io mailing lists](https://groups.io/g/corda-dev/message/190?p=,,,20,0,0,0::relevance,,updates+distribution,20,2,0,22686107):
+
+* BNO should be able to notify members about new CorDapp version availability
+* BNO should be able to revoke a specific version of CorDapp from their BN
```

Comment:
what does it mean to "revoke" a CorDapp?

Reply:
thanks for the clarification.  In which case, I think we should reword the requirement...  to something like: "BNO should be able to notify members that they a specific version of a CorDapp should no longer be considered valid, and members should be able to configure their nodes to use such notifications to prevent such applications from executing" (is that the right wording though?  ie is this about stopping a node from running the app?  or about having an node know it should reject transactions that reference a particular version of the code?)

Reply:
Makes sense. Added clarification about what revocation in this context actually means.